src/registries/npm-registry.js

```diff
@@ -139,7 +138,7 @@ export default class NpmRegistry extends Registry {
         await fs.mkdirp(mirrorLoc);
       }
 
-      defaults(this.config, config);
+      this.config = Object.assign({}, config, this.config);
```

Comment:
I understand that defaults mutates the object instead of creating a copy.
Would it be a problem when passing a particular property across modules?